src/ProjectSystemTools/BuildLogging/Model/Frontend/FrontendBuildTableDataSource.cs

```diff
@@ -50,39 +47,39 @@ public ITableManager Manager
             }
         }
 
-        public BuildTableDataSource()
+        [ImportingConstructor]
+        public FrontEndBuildTableDataSource(IBuildLoggerService loggerService)
+        {
+            _loggerService = loggerService;
+            SupportRoslynLogging = _loggerService.SupportsRoslynLogging();
+        }
+
+        public bool IsLogging
         {
-            _evaluationLogger = new EvaluationLogger(this);
-            _roslynLogger = new RoslynLogger(this);
+            get
+            {
+                return _loggerService.IsLogging();
+            }
         }
 
         public void Start()
         {
-            IsLogging = true;
-            ProjectCollection.GlobalProjectCollection.RegisterLogger(_evaluationLogger);
-            _roslynLogger.Start();
+            _loggerService.Start(UpdateEntries);
         }
 
         public void Stop()
         {
-            IsLogging = false;
-            ProjectCollection.GlobalProjectCollection.UnregisterAllLoggers();
-            _roslynLogger.Stop();
+            _loggerService.Stop();
         }
 
         public void Clear()
         {
-            foreach (var build in _entries)
-            {
-                build.Dispose();
-            }
-            _entries = ImmutableList<Build>.Empty;
+            _loggerService.Clear();
+            _entries = ImmutableList<UIBuildSummary>.Empty;
             CurrentVersionNumber++;
             NotifyChange();
```

Comment:
Not related to your changes, but `NotifyChange` already increments `CurrentVersionNumber`. Therefore `Clear` increments the version twice.